tools/create-genesis/src/genesis.rs

```diff
@@ -0,0 +1,235 @@
+use crate::common::{clean_0x, string_2_bytes};
+use crate::contracts::ContractsData;
+use crate::miner::Miner;
+use crate::params::InitData;
+use crate::solc::Solc;
+use ethabi::Contract;
+use evm::cita_types::U256;
+use json;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+use std::str::FromStr;
+
+pub struct GenesisCreator<'a> {
+    pub contract_dir: &'a str,
+    pub contract_docs_dir: &'a str,
+    pub genesis_path: &'a str,
+    pub timestamp: u64,
+    pub init_token: &'a str,
+    pub prevhash: &'a str,
+    pub contract_args: InitData,
+    pub contract_list: ContractsData,
+    pub accounts: BTreeMap<String, Account>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Account {
+    pub nonce: U256,
+    pub code: String,
+    pub storage: BTreeMap<String, String>,
+    pub value: U256,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Genesis {
+    pub timestamp: u64,
+    pub prevhash: String,
+    pub alloc: BTreeMap<String, Account>,
+}
+
+impl Default for Genesis {
+    fn default() -> Self {
+        Genesis {
+            timestamp: 0,
+            prevhash: String::default(),
+            alloc: BTreeMap::new(),
+        }
+    }
+}
+
+impl<'a> GenesisCreator<'a> {
+    pub fn new(
+        contract_dir: &'a str,
+        contract_docs_dir: &'a str,
+        params_path: &'a str,
+        genesis_path: &'a str,
+        timestamp: u64,
+        init_token: &'a str,
+        prevhash: &'a str,
+    ) -> Self {
+        let params = InitData::load_contract_args(params_path);
+        let contracts_list = contract_dir.to_owned() + "/contracts.yml";
+        let constracts = ContractsData::load_contract_list(&contracts_list);
+
+        GenesisCreator {
+            contract_dir,
+            contract_docs_dir,
+            genesis_path,
+            timestamp,
+            init_token,
+            prevhash,
+            contract_args: params,
+            contract_list: constracts,
+            accounts: BTreeMap::new(),
+        }
+    }
+
+    pub fn create(&mut self) {
+        // 1. Check compile exit or not
+        if !Solc::compiler_version() {
+            panic!("solc compiler not exit");
+        }
+        // 2. Init normal contracts
+        self.init_normal_contracts();
+        // 3. Init permission contracts
+        self.init_permission_contracts();
+        // 4. Save super admin
+        let super_admin = self.contract_args.contracts.admin.admin.clone();
+        self.set_account_value(
+            &super_admin,
+            U256::from_str(clean_0x(&self.init_token)).unwrap(),
+        );
+        // 5. Save genesis to file
+        self.save_to_file();
+        println!("Create genesis successfully !");
+    }
+
+    pub fn init_normal_contracts(&mut self) {
```

Comment:
And check the function `init_permission_contracts`,
Could we refactor them?

Reply:
Another pr later.